tests/phpunit/tests/pomo/po.php

```diff
@@ -35,178 +35,297 @@ function setUp() {
 "\n"
 "--The ClassicPress Team\n"
 "http://wordpress.org/\n"';
-		$this->a90 = str_repeat("a", 90);
-		$this->po_a90 = "\"$this->a90\"";
-    }
+		$this->a90     = str_repeat( 'a', 90 );
+		$this->po_a90  = "\"$this->a90\"";
+	}
 
 	function test_prepend_each_line() {
 		$po = new PO();
-		$this->assertEquals('baba_', $po->prepend_each_line('', 'baba_'));
-		$this->assertEquals('baba_dyado', $po->prepend_each_line('dyado', 'baba_'));
-		$this->assertEquals("# baba\n# dyado\n# \n", $po->prepend_each_line("baba\ndyado\n\n", '# '));
+		$this->assertEquals( 'baba_', $po->prepend_each_line( '', 'baba_' ) );
+		$this->assertEquals( 'baba_dyado', $po->prepend_each_line( 'dyado', 'baba_' ) );
+		$this->assertEquals( "# baba\n# dyado\n# \n", $po->prepend_each_line( "baba\ndyado\n\n", '# ' ) );
 	}
 
 	function test_poify() {
 		$po = new PO();
 		//simple
-		$this->assertEquals('"baba"', $po->poify('baba'));
+		$this->assertEquals( '"baba"', $po->poify( 'baba' ) );
 		//long word
-		$this->assertEquals($this->po_a90, $po->poify($this->a90));
+		$this->assertEquals( $this->po_a90, $po->poify( $this->a90 ) );
 		// tab
-		$this->assertEquals('"ba\tba"', $po->poify("ba\tba"));
+		$this->assertEquals( '"ba\tba"', $po->poify( "ba\tba" ) );
 		// do not add leading empty string of one-line string ending on a newline
-		$this->assertEquals('"\\\\a\\\\n\\n"', $po->poify("\a\\n\n"));
+		$this->assertEquals( '"\\\\a\\\\n\\n"', $po->poify( "\a\\n\n" ) );
 		// backslash
-		$this->assertEquals('"ba\\\\ba"', $po->poify('ba\\ba'));
+		$this->assertEquals( '"ba\\\\ba"', $po->poify( 'ba\\ba' ) );
 		// random wordpress.pot string
 		$src = 'Categories can be selectively converted to tags using the <a href="%s">category to tag converter</a>.';
-		$this->assertEquals("\"Categories can be selectively converted to tags using the <a href=\\\"%s\\\">category to tag converter</a>.\"", $po->poify($src));
+		$this->assertEquals( '"Categories can be selectively converted to tags using the <a href=\\"%s\\">category to tag converter</a>."', $po->poify( $src ) );
```

Comment:
Need confirmation that the expected result is supposed to have a slash before the double quotes.

Reply:
On first look this seems correct. Because the original used only double quotes, each subsequent use of the double quote within the string needs escaped with a slash.

By converting the opening and closing quotes to single quotes, all of the contains double quotes can be unslashed. The `href` complicates things for sure but the unslashing only really impacts the quotes in an `href=""`.

Reply:
I put both versions in a little script and they echo out the same, plus the compare says equal. Thanks for looking.